Packs/Microsoft365Defender/Integrations/Microsoft365DefenderEventCollector/Microsoft365DefenderEventCollector.py

```
@@ -0,0 +1,354 @@
+                return stored[:self.options.limit]
+        return stored
+
+    def call(self) -> requests.Response:
+        return self.client.call(self.client.request)
+
+    @staticmethod
+    @abstractmethod
+    def get_last_run(events: list) -> dict:
+        """Logic to get the last run from the events
+        Example:
+        """
+        return {'after': events[-1]['created']}
+
+    @abstractmethod
+    def _iter_events(self):
+        """Create iterators with Yield"""
+        pass
+
+# END COPY OF SiemApiModule
+
+
+class DefenderIntegrationOptions(IntegrationOptions):
+    first_fetch: str
+
+
+class DefenderAuthenticator(BaseModel):
+    verify: bool
+    url: str
+    tenant_id: str
+    client_id: str
+    credentials: dict
+    ms_client: Any = None
+
+    def set_authorization(self, request: IntegrationHTTPRequest):
+        try:
+            if not self.ms_client:
+                demisto.debug('try init the ms client for the first time')
+                self.ms_client = MicrosoftClient(
+                    base_url=self.url,
+                    tenant_id=self.tenant_id,
+                    auth_id=self.client_id,
+                    enc_key=self.credentials.get('password'),
+                    scope=SECURITY_SCOPE,
+                    verify=self.verify,
+                    self_deployed=True
+                )
+
+            token = self.ms_client.get_access_token()
+            auth = {'Authorization': f'Bearer {token}'}
+            if request.headers:
+                request.headers |= auth  # type: ignore[assignment, operator]
+            else:
+                request.headers = auth  # type: ignore[assignment]
+
+            demisto.debug('getting access token for Defender Authenticator - succeeded')
+
+        except BaseException:
+            # catch BaseException to catch also sys.exit via return_error
+            err_msg = 'Fail to authenticate with Microsoft services, see the error details in the log'
```

Don't hide the original error.
Write it to the log.